### ISSUE TITLE
Look into adding ROI support to the GUI similar to how you can use it in HEXRD

### ISSUE THREAD

**Issue by psavery** [HEXRD/hexrdgui#1446]
This goes along with #1347

We will hopefully get an example soon.

**Comment by psavery** [comment]
More info:

HEXRD supports an ROI parameter in the detector parameters of the instrument config. This defines a "region of interest", so that the detector only uses a region of pixels from the image, like so:

```yaml
detectors:
  ff1_0_0:
    pixels:
      roi:
      - 0
      - 0
  # ...
  ff1_0_1:
    pixels:
      roi:
      - 0
      - 1536
```

(those are start pixels for row/column, and then the number of rows/columns in the detector settings is used to determine where to stop)

Then in the script, multiple detectors are specified for a single image file, like so:

```yaml
image_series:
  format: frame-cache
  data:
    - file: mruby-1104-1_7_ff1_000011-cachefile.npz
      panel:  # must match detector keys
        - ff1_0_0
        - ff1_0_1
        # ...
```

The question now is: should we add support for this in the GUI?

My opinion so far: I think we need to discuss the pros/cons and alternative solutions before going down this route. Having ROI support in the command line interface makes sense and keeps it flexible (although it may better belong in the image series options rather than in the detector options). But adding this to the GUI exactly as it is may be complicated, and it may be somewhat non-intuitive.

In my limited testing, it looks like there are many places in the GUI where we would have to add logic to handle it, and thus could be a fair amount of work to add. It would also add an extra degree of maintenance (if we make any changes, we'd have to ensure it doesn't break ROI support), which could make maintenance more difficult over time. But we can certainly still do it anyways if it is important.

If we could alternatively add an option in the GUI to break apart detectors into multiple sub-detectors (which would also break up the image files into separate image files), that may help alleviate the issue. Perhaps also we could allow instrument configs with ROI to be loaded in, then immediately have the user select image files, and then immediately break up the image files into separate files and eliminate the ROI section in the config. That would be easy to add, and easy to maintain. Maybe this would only pose a problem if the file sizes were very large.

So to summarize: we could add ROI support (where an `roi` parameter is maintained in the instrument config), but it may be a fair amount of work, and I think there are some good alternatives that may be easier to support and maintain (which could include loading in an instrument file with an `roi` parameter and then converting it to something easier to work with). What do you think, @donald-e-boyce?
